### PR TITLE
Branch movie

### PR DESCRIPTION
**Pull request by santiagozeballos28**

### CODE REVIEW

File: src/main/java/com/trueffect/model/Person.java [hunk 1]
```diff
@@ -84,8 +85,8 @@ public void setBirthday(String birthday) {
 
     
     @Override
-     public ModelObject insertResourse(int idCreatorUser)throws Exception{
-     return PersonCrud.insertrenterUser(idCreatorUser,this);
+     public ModelObject insertResourse(int idCreatorUser, Connection connection)throws Exception{
+     return PersonCrud.insertrenterUser(connection,idCreatorUser, this);
```
Comment (cmtr0704) [inline src/main/java/com/trueffect/model/Person.java]: for names of class , objects or variables please use the style  camelCase

File: src/main/java/com/trueffect/sql/crud/JobCrud.java [hunk 2]
```diff
@@ -11,13 +11,13 @@
 public class JobCrud {
     public static Job getJobOf(int id)throws Exception {
             Job job=null;
-            PostgresSQLConnection.connectionDB();
+            DatabasePostgres.getConection();
             try {
                           String sql = "SELECT id, name_job\n" +
```
Comment (cmtr0704) [inline src/main/java/com/trueffect/sql/crud/JobCrud.java]: Please align the query using the format RIVER .Use as example this url :http://www.sqlstyle.guide/#white-space

File: src/main/java/com/trueffect/sql/crud/JobCrud.java [hunk 3]
```diff
@@ -11,13 +11,13 @@
...
                              "  FROM data_job, job\n" +
                              "  WHERE data_job.id_job= job.id AND id_person=?;" ;
 
-                PreparedStatement st = PostgresSQLConnection.connection.prepareStatement(sql);
+                PreparedStatement st = DatabasePostgres.connection.prepareStatement(sql);
                 st.setInt(1, id);
                 ResultSet rs = st.executeQuery();
             if (rs.next()) {
```
Comment (cmtr0704) [inline src/main/java/com/trueffect/sql/crud/JobCrud.java]: Please do the documentation in English line 25

File: src/main/java/com/trueffect/sql/crud/PersonCrud.java [hunk 4]
```diff
@@ -40,26 +42,33 @@ public static Person insertrenterUser(int idJob, Person renterUser) throws Excep
                 " VALUES ('"+typeIdentifier+"','"+identifier+"','"+lastName+"', '"
                         +firstName+"','"+ genre+"','"+birthay+"','2017-12-08','"+idJob+"',null,null,'Active')";
       
-                consulta.execute(sql);
+                query.execute(sql);
                 String msg = "The user " + firstName + " " + lastName + " has been successfully inserted";
```
Comment (cmtr0704) [inline src/main/java/com/trueffect/sql/crud/PersonCrud.java]: I dont found the usage  of the variable "msg "

File: src/main/java/com/trueffect/sql/crud/PersonCrud.java [hunk 5]
```diff
@@ -40,26 +42,33 @@ public static Person insertrenterUser(int idJob, Person renterUser) throws Excep
...
                 renterUserInserted = getByTypeIdentifier(typeIdentifier, identifier);
               } catch (Exception e) {
                 throw new ErrorResponse(CodeStatus.INTERNAL_SERVER_ERROR, e.getMessage());
-            }
-            PostgresSQLConnection.closeDB();
+            }finally{
+                 try {
```
Comment (cmtr0704) [inline src/main/java/com/trueffect/sql/crud/PersonCrud.java]: You have methods for close and create conection, can you create a method for close  sql and put in the respective class

File: src/main/java/com/trueffect/logica/PersonLogic.java [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+
+    public Person createPerson(int id, Person person, DataCondition conditiondata) throws Exception {
+        Person personRes = null;
+        ErrorContainer errorContainer = new ErrorContainer();
+        //open conection 
+        Connection connection = DataBasePostgres.getConection();
+        try {
+            //Validation of data 
+            conditiondata.complyCondition(person);
+            PersonValidationsDB.veriryDataInDataBase(connection, person);
```
Comment (cmtr0704) [inline src/main/java/com/trueffect/logica/PersonLogic.java]: Change the name of the method :+        veriryDataInDataBase .Wath is the objective of thie method?

File: src/main/java/com/trueffect/logica/PersonLogic.java [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+            personRes = PersonCrud.insertrenterUser(connection, id, person);
```
Comment (cmtr0704) [inline src/main/java/com/trueffect/logica/PersonLogic.java]: Use Format camelCase for names object and variables.  ---->   insertrenterUser

File: src/main/java/com/trueffect/logica/PersonLogic.java [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+            connection.commit();
+        } catch (ErrorResponse errorResponse) {
+            errorContainer.addError(new ErrorResponse(errorResponse.getCode(), errorResponse.getMessage()));
+            OperationDataBase.connectionRollback(connection, errorContainer);
+        } finally {
+            OperationDataBase.connectionClose(connection, errorContainer);
+        }
+        if (errorContainer.size() > 0) {
```
Comment (cmtr0704) [inline src/main/java/com/trueffect/logica/PersonLogic.java]: Use the method isEmpty to validate if exist errors

File: src/main/java/com/trueffect/conection/db/OperationDataBase.java [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+
+/**
+ *
+ * @author santiago.mamani
+ */
+public class OperationDataBase {
+
+    public static void connectionClose(Connection connection, ErrorContainer errorContainer) {
+
+        if (connection != null) {
```
Comment (cmtr0704) [inline src/main/java/com/trueffect/conection/db/OperationDataBase.java]: If connection is null ,what will be  happen.This method will be called just when you have a open conection

File: src/main/java/com/trueffect/conection/db/OperationDataBase.java [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+            try {
+                connection.close();
+            } catch (SQLException ex) {
+                errorContainer.addError(new ErrorResponse(CodeStatus.INTERNAL_SERVER_ERROR, ex.getMessage()));
+            }
+        }
+    }
+
+    public static void connectionRollback(Connection connection, ErrorContainer errorContainer) {
+        if (connection != null) {
```
Comment (cmtr0704) [inline src/main/java/com/trueffect/conection/db/OperationDataBase.java]: Same ..

File: src/main/java/com/trueffect/logica/Generator.java [hunk 11]
```diff
@@ -0,0 +1,55 @@
+package com.trueffect.logica;
+
+import com.trueffect.model.Person;
+
+/**
+ *
+ * @author santiago.mamani
+ */
```
Comment (cmtr0704) [inline src/main/java/com/trueffect/logica/Generator.java]: The name is generic,please be more specifc

File: src/main/java/com/trueffect/logica/Generator.java [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+                resSet = resSet + "genre= '" + varSet + "',";
+            }
+        }
+        varSet = person.getBirthday();
+        if (varSet != null) {
+            if (varSet.length() > 0) {
+                resSet = resSet + "birthday= '" + varSet + "',";
+            }
+        }
+
```
Comment (cmtr0704) [inline src/main/java/com/trueffect/logica/Generator.java]: Remove spaces

File: src/main/java/com/trueffect/logica/Generator.java [hunk 13]
```diff
@@ -0,0 +1,55 @@
...
+import com.trueffect.model.Person;
+
+/**
+ *
+ * @author santiago.mamani
+ */
+public class Generator {
+
+    public static String getStringSet(Person person) {
```
Comment (cmtr0704) [inline src/main/java/com/trueffect/logica/Generator.java]: What is the objective of this class? please  I need your point

File: src/main/java/com/trueffect/logica/PersonLogic.java [hunk 14]
```diff
@@ -0,0 +1,116 @@
...
+            verifyPerson(connection, idPerson);
+            res = PersonCrud.deleteById(connection, idPerson, idUserModify);
+            connection.commit();
+        } catch (ErrorResponse e) {
+            errorContainer.addError(new ErrorResponse(e.getCode(), e.getMessage()));
+            OperationDataBase.connectionRollback(connection, errorContainer);
+        } finally {
+            OperationDataBase.connectionClose(connection, errorContainer);
+        }
+        if (errorContainer.size() > 0) {
```
Comment (cmtr0704) [inline src/main/java/com/trueffect/logica/PersonLogic.java]: Use the method isEmpty to validate if exist errors

File: src/main/java/com/trueffect/logica/PersonLogic.java [hunk 15]
```diff
@@ -0,0 +1,116 @@
...
+
+    private static void verifyPerson(Connection connection, int idPerson) throws Exception {
+        Person person = PersonCrud.getPerson(connection, idPerson);
+        if (person == null) {
+            throw new ErrorResponse(CodeStatus.NOT_FOUND, Message.NOT_RESOURCE);
+        }
+    }
+
+    public Person update(Person person, int idRenter, int idUserModify) throws Exception {
+        Person personRes = null;
```
Comment (cmtr0704) [inline src/main/java/com/trueffect/logica/PersonLogic.java]: Remove this : Person personRes;
is better do this : Person personRes = PersonCrud.updateRenterUser(connection, idRenter, idUserModify, setString

File: src/main/java/com/trueffect/logica/PersonLogic.java (hunk 7 above)
Comment (cmtr0704) [inline src/main/java/com/trueffect/logica/PersonLogic.java]: Please remove this : Person personRes=null changed to :
Person  personRes = PersonCrud.insertrenterUser(connection, id, person);

File: src/main/java/com/trueffect/logica/PersonLogic.java [hunk 16]
```diff
@@ -0,0 +1,116 @@
...
+    }
+
+    private static void verifyPerson(Connection connection, int idPerson) throws Exception {
+        Person person = PersonCrud.getPerson(connection, idPerson);
+        if (person == null) {
+            throw new ErrorResponse(CodeStatus.NOT_FOUND, Message.NOT_RESOURCE);
+        }
+    }
+
+    public Person update(Person person, int idRenter, int idUserModify) throws Exception {
```
Comment (cmtr0704) [inline src/main/java/com/trueffect/logica/PersonLogic.java]: Please add the validation that the path param id be the same that object Person

File: src/main/java/com/trueffect/logica/PersonLogic.java [hunk 17]
```diff
@@ -0,0 +1,116 @@
...
+        Person personRes = null;
+        ErrorContainer errorContainer = new ErrorContainer();
+        //open conection 
+        Connection connection = DataBasePostgres.getConection();
+        try {
+            //Validation of data             
+            verifyModifierUser(connection, idUserModify);
+            Job job = JobCrud.getJobOf(connection, idUserModify);
```
Comment (cmtr0704) [inline src/main/java/com/trueffect/logica/PersonLogic.java]: Please add documentation about the usage of RenterUserUpdate

File: src/main/java/com/trueffect/logica/PersonLogic.java [hunk 18]
```diff
@@ -0,0 +1,116 @@
...
+            RenterUserUpdate rentUserUpdate = new RenterUserUpdate();
+            rentUserUpdate.setIdUserModify(job.getNameJob());
```
Comment (cmtr0704) [inline src/main/java/com/trueffect/logica/PersonLogic.java]: Are you setting the name in a IDuserModify this is correct  ? are the same type ?

File: src/main/java/com/trueffect/logica/PersonLogic.java [hunk 19]
```diff
@@ -0,0 +1,116 @@
...
+            personRes = PersonCrud.updateRenterUser(connection, idRenter, idUserModify, setString);
+            connection.commit();
+        } catch (ErrorResponse exception) {
+            errorContainer.addError(new ErrorResponse(exception.getCode(), exception.getMessage()));
+            OperationDataBase.connectionRollback(connection, errorContainer);
+        } finally {
+            OperationDataBase.connectionClose(connection, errorContainer);
+        }
+        if (errorContainer.size() > 0) {
+            throw new ErrorResponse(errorContainer.getCodeStatusEnd(), errorContainer.allMessagesError());
```
Comment (cmtr0704) [inline src/main/java/com/trueffect/logica/PersonLogic.java]: Use you method isEmpty for set the errors

File: src/main/java/com/trueffect/logica/PersonValidationsDB.java [hunk 20]
```diff
@@ -0,0 +1,73 @@
...
+ * @author santiago.mamani
+ */
+public class PersonValidationsDB {
+
+    /*
+    * For dates duplidates
+     */
+    public static void veriryDataInDataBase(Connection connection, Person personNew) throws Exception {
+        ErrorContainer errorContainer = new ErrorContainer();
+        Person person = (Person) PersonCrud.getPersonByIdentifier(connection, personNew.getTypeIdentifier(), personNew.getIdentifier());
```
Comment (cmtr0704) [inline src/main/java/com/trueffect/logica/PersonValidationsDB.java]: Why the reason that you cast the response of PersonCrud.getPersonByIdentifier......

File: src/main/java/com/trueffect/logica/PersonValidationsDB.java [hunk 21]
```diff
@@ -0,0 +1,73 @@
...
+        }
+        person = (Person) PersonCrud.getPersonByName(connection, personNew.getLastName(), person.getFirstName());
+        if (person != null) {
+            errorContainer.addError(new ErrorResponse(CodeStatus.BAD_REQUEST, Message.THE_NAMES_ALREADY_EXIST));
+        }
+        if (!errorContainer.isEmpty()) {
+            throw new ErrorResponse(errorContainer.getCodeStatusEnd(), errorContainer.allMessagesError());
+        }
+    }
+
```
Comment (cmtr0704) [inline src/main/java/com/trueffect/logica/PersonValidationsDB.java]: Please  read this article http://javarevisited.blogspot.com/2013/07/when-to-make-method-static-in-java.html.When use a static method

File: src/main/java/com/trueffect/messages/Message.java [hunk 22]
```diff
@@ -1,24 +1,39 @@
...
-    public static  String NOT_VALID_IDENTIFIER="The dentifier not's  valid";     
-    public static  String DUPLICATE_IDENTIFIER="The identifier already exists";     
-    public static  String NOT_MEET_THE_AGE="To be registered must be greater than 15 years"; 
-    /*
-    * constant for resources
-    */
-    public static  String NOT_FOUND="The requested resource does not exist";
+
+    // Message of Empty
+    public static String EMPTY_TYPE_IDENTIFIER = "The type of identifier is required";
```
Comment (cmtr0704) [inline src/main/java/com/trueffect/messages/Message.java]: In a different story we unify the messages with similar text

File: src/main/java/com/trueffect/messages/Message.java [hunk 23]
```diff
@@ -1,24 +1,39 @@
...
+    public static String NOT_VALID_FIRST_NAME = "The first name is not valid";
+    public static String NOT_VALID_GENRE = "The genre is not valid";
+    public static String NOT_VALID_BIRTHDAY = "The date format is not valid";
+    public static String NOT_SAME_TYPE = "The identifier is not the same type";
+    //Message de restrictions    
+    public static String SIZE_IDENTIFIER = "The identifier is very long, must be less than 10 characters";
+    public static String SIZE_LAST_NAME = "The last name is very long, must be less than 50 characters";
+    public static String SIZE_FIRST_NAME = "The first name is very long, must be less than 50 characters";
+    public static String DUPLICATE_IDENTIFIER = "The identifier already exists";
+    public static String NOT_MEET_THE_AGE = "To be registered must be greater than 15 years";
```
Comment (cmtr0704) [inline src/main/java/com/trueffect/messages/Message.java]: This message NOT_MEET_THE_AGE , for who is applied?  please add in the text:" To be registered must be greater than 15 years"

File: src/main/java/com/trueffect/messages/Message.java [hunk 24]
```diff
@@ -1,24 +1,39 @@
...
+    public static String THE_NAMES_ALREADY_EXIST = "The names already exist";
+    public static String NOT_VALID_AGE = "Must be  at least 15 years old";
```
Comment (cmtr0704) [inline src/main/java/com/trueffect/messages/Message.java]: Please let me now the objective of : NOT_MEET_THE_AGE , NOT_VALID_AGE

File: src/main/java/com/trueffect/messages/Message.java [hunk 25]
```diff
@@ -1,24 +1,39 @@
...
+    // Message of results
+    public static String NOT_FOUND = "The requested resource does not exist";
+    public static String NOT_RESOURCE = "There is not resource";
+    // Message of permission
+    public static String NOT_HAVE_PERMISSION_TYPE_IDENTIFIER = "You do not have permission to modify type identifier";
```
Comment (cmtr0704) [inline src/main/java/com/trueffect/messages/Message.java]: I think is better this :" The user don't have permission to modify type identifier" , let me know if you agree  or not with that

File: src/main/java/com/trueffect/logica/Generator.java (hunk 13 above)
Comment (santiagozeballos28) [reply src/main/java/com/trueffect/logica/Generator.java]: To generate sql either to modify or for conditions. Example to modify would generate a SET in the following way... "type_ indedifier=CI, identifier=12344456, lastName='Zeballos Arnez'"

File: src/main/java/com/trueffect/sql/crud/PersonCrud.java [hunk 26]
```diff
@@ -1,187 +1,177 @@
...
-                String msg = "The user " + firstName + " " + lastName + " has been successfully inserted";
-                renterUserInserted = getByTypeIdentifier(typeIdentifier, identifier);
-              } catch (Exception e) {
-                throw new ErrorResponse(CodeStatus.INTERNAL_SERVER_ERROR, e.getMessage());
+    public static Person insertrenterUser(Connection connection, int idJob, Person renterUser) throws Exception {
+        Person renterUserInserted = null;
+        Statement query = null;
+        try {
+            String typeIdentifier = renterUser.getTypeIdentifier().trim();
+            String identifier = renterUser.getIdentifier().trim();
```
Comment (cmtr0704) [inline src/main/java/com/trueffect/sql/crud/PersonCrud.java]: You have this method  Person(int id, String typeIdentifier, String identifier, String lastName, String firstName, String genre, String birthday)  that to the trim of all values .This code be unnecesary

File: src/main/java/com/trueffect/sql/crud/PersonCrud.java [hunk 27]
```diff
@@ -1,187 +1,177 @@
...
+    public static Person getPersonByIdentifier(Connection connection, String typeIdentifier, String identifier) throws Exception {
+        Person person = null;
         try {
-            PostgresSQLConnection.connectionDB();
-            try {
-                renterUser = getRenterUserById(id);
-                } catch (Exception e) {
-                throw new ErrorResponse(CodeStatus.NOT_FOUND, e.getMessage());
+            Statement consulta = (Statement) connection.createStatement();
+            String sql = "SELECT id, type_identifier, identifier, last_name, first_name, genre, birthday\n"+
```
Comment (cmtr0704) [inline src/main/java/com/trueffect/sql/crud/PersonCrud.java]: please aligiment the query

File: src/main/java/com/trueffect/sql/crud/PersonCrud.java [hunk 28]
```diff
@@ -1,187 +1,177 @@
...
-        st.setInt(1, id);
-        ResultSet rs = st.executeQuery();
-        if (rs.next()) {
-           renterUser = new Person(rs.getInt("id"), rs.getString("type_identifier"), rs.getString("identifier"), rs.getString("last_name"), rs.getString("first_name"), rs.getString("genre"), rs.getString("birthday"));
-        } else {
-            throw new ErrorResponse(CodeStatus.NOT_FOUND, Message.NOT_FOUND);
+    public static Person getPersonByName(Connection connection, String lastName, String firstName) throws Exception {
+        Person person = null;
+        try {
+            String sqlGet = "SELECT id, type_identifier, identifier, last_name, first_name, genre, birthday\n"+
```
Comment (cmtr0704) [inline src/main/java/com/trueffect/sql/crud/PersonCrud.java]: same here

File: src/main/java/com/trueffect/sql/crud/PersonCrud.java [hunk 29]
```diff
@@ -1,187 +1,177 @@
...
+    public static Person getPerson(Connection connection, int idPerson) throws Exception {
+        Person person = null;
         try {
- PostgresSQLConnection.connectionDB();
-        PreparedStatement st = PostgresSQLConnection.connection.prepareStatement(sql);
-        ResultSet rs = st.executeQuery();
-        
-        if (rs.next()) {
-           renterUser = new Person(rs.getInt("id"), 
+            String sql = "SELECT id,type_identifier,identifier, last_name, first_name, genre, birthday\n"+
```
Comment (cmtr0704) [inline src/main/java/com/trueffect/sql/crud/PersonCrud.java]: same here

File: src/main/java/com/trueffect/sql/crud/PersonCrud.java [hunk 30]
```diff
@@ -1,187 +1,177 @@
...
-            PostgresSQLConnection.closeDB();
-        } catch (Exception e) {
-            throw new ErrorResponse(CodeStatus.INTERNAL_SERVER_ERROR, e.getMessage());
+    public static Person deleteById(Connection connection, int idPerson, int idUserModifier) throws Exception {
+        Person person = null;
+        try {
+            person = getPerson(connection, idPerson);
+            String sql
+                    = "UPDATE PERSON\n"+
+                      "   SET status='Inactive', date_modifier =  current_date , user_modifier= ?"+
```
Comment (cmtr0704) [inline src/main/java/com/trueffect/sql/crud/PersonCrud.java]: same here

File: src/main/java/com/trueffect/sql/crud/PersonCrud.java [hunk 31]
```diff
@@ -1,187 +1,177 @@
...
-                   }     
-            } catch (Exception e) {
-                throw new ErrorResponse(CodeStatus.NOT_FOUND, e.getMessage());
-            }
-        return renterUserPerson;
-    }
-      public static int getIdOf(String lastName,String firstName) throws Exception{
-       int idPerson=0;
+    public static Person updateRenterUser(Connection connection, int idPerson, int idUserModifier,String setString) throws Exception {
+        Person person = null;
```
Comment (cmtr0704) [inline src/main/java/com/trueffect/sql/crud/PersonCrud.java]: Is better create the object like this :  Person person =getPerson(....).. @

File: src/main/java/com/trueffect/validation/PersonValidation.java [hunk 32]
```diff
@@ -1,75 +1,105 @@
...
+        return name.length() <= size;
     }
-  
-    public static boolean isValidGenre(String genre) {
-        Genre genreEnum= null;
-        if (genre != null) {
-            return genre.equals(genreEnum.M.name()) || genre.equals(genreEnum.F.name());
+
+    public static boolean isValidLastName(String lastName) {
+        if (lastName.contains(" ")) {
```
Comment (cmtr0704) [inline src/main/java/com/trueffect/validation/PersonValidation.java]: you can use this StringUtils.isBlank() --> https://stackoverflow.com/questions/23419087/stringutils-isblank-vs-string-isempty

Comment (cmtr0704) [inline src/main/java/com/trueffect/validation/PersonValidation.java]: why you use this two patters .LAST_NAME_TWO,LAST_NAME_ONE , wath is the difference?

File: src/main/java/com/trueffect/validation/PersonValidation.java [hunk 33]
```diff
@@ -1,75 +1,105 @@
...
+            return Pattern.matches(RegularExpression.LAST_NAME_TWO, lastName);
+        } else {
+            return Pattern.matches(RegularExpression.LAST_NAME_ONE, lastName);
         }
-        return false;
+    }
+
+    public static boolean isValidGenre(String genre) {
+        Genre genreEnum = null;
```
Comment (cmtr0704) [inline src/main/java/com/trueffect/validation/PersonValidation.java]: You have a enum plese use for the comparation

File: src/main/java/com/trueffect/validation/PersonValidation.java [hunk 34]
```diff
@@ -1,75 +1,105 @@
...
-        return false;
+        return Pattern.matches(RegularExpression.DATE, date);
     }
-   
 
-    public static boolean isValidIdentifier(String typeIdentifier, String identifier)  throws Exception{
-     return true;
+    public static boolean isValidAge(String age) {
+        boolean res = false;
+        Date date = new Date();
```
Comment (cmtr0704) [inline src/main/java/com/trueffect/validation/PersonValidation.java]: Please use the library of java  get the year and recibe as parameter the age to validate

File: src/main/java/com/trueffect/validation/RenterUserUpdate.java [hunk 35]
```diff
@@ -1,45 +1,107 @@
-
+/*
```
Comment (cmtr0704) [inline src/main/java/com/trueffect/validation/RenterUserUpdate.java]: remove this comment

File: src/main/java/com/trueffect/validation/PersonValidation.java (hunk 32 above)
Comment (santiagozeballos28) [reply src/main/java/com/trueffect/validation/PersonValidation.java]: LAST_NAME_TWO is to validate two surnames and LAST-NAME_ONE is to validate a single surname.

**Pull request closed without merging**